File: app/src/main/java/com/example/sarahn/waterreminderapp/classes/ClsRestartAlarm.java
package com.example.sarahn.waterreminderapp.classes;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.support.v4.content.WakefulBroadcastReceiver;
import android.widget.Toast;

import com.example.sarahn.waterreminderapp.Utils.Logging;
import com.example.sarahn.waterreminderapp.Utils.SharedPrefUtils;

/**
 * Created by SarahN on 7/3/2017.
 */
public class ClsRestartAlarm extends WakefulBroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {

        if ("android.intent.action.BOOT_COMPLETED".equals(intent.getAction())) {

            SharedPrefUtils.setIsBoot(context, true);
            Intent i = new Intent(context, BootService.class);
            context.startService(i);
        }
    }
}

File: app/src/main/java/com/example/sarahn/waterreminderapp/classes/ClsRestartAlarm.kt
package com.example.sarahn.waterreminderapp.classes

import android.content.Context
import android.content.Intent
import androidx.legacy.content.WakefulBroadcastReceiver
import com.example.sarahn.waterreminderapp.Utils.SharedPrefUtils

/**
 * Created by SarahN on 7/3/2017.
 */
class ClsRestartAlarm : WakefulBroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if ("android.intent.action.BOOT_COMPLETED" == intent.action) {
            SharedPrefUtils.setIsBoot(context, true)
            val i = Intent(context, BootService::class.java)
            context.startService(i)
        }
    }
}